doc/specs/#2046 - Unified keybindings and commands, and synthesized action names.md

```diff
@@ -0,0 +1,589 @@
+
+These changes to the `defaults.json` are represented in json as the following:
+
+```json
+"commands": [
+  {
+    "icon": null,
+    "name": { "key": "NewTabWithProfileRootCommandName" },
+    "commands": [
+      {
+        "iterateOn": "profiles",
+        "icon": "${profile.icon}",
+        "name": "${profile.name}",
+        "command": { "action": "newTab", "profile": "${profile.name}" }
+      }
+    ]
+  },
+  {
+    "icon": null,
+    "name": { "key": "SelectColorSchemeRootCommandName" },
+    "commands": [
+      {
+        "iterateOn": "schemes",
+        "icon": null,
+        "name": "${scheme.name}",
+        "command": { "action": "selectColorScheme", "scheme": "${scheme.name}" }
+      }
+    ]
+  },
+  {
+    "icon": null,
+    "name": { "key": "SplitPaneWithProfileRootCommandName" },
+    "commands": [
+      {
+        "iterateOn": "profiles",
+        "icon": "${profile.icon}",
+        "name": { "key": "SplitPaneWithProfileCommandName" },
+        "commands": [
+          {
+            "icon": null,
+            "name": { "key": "SplitPaneName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "automatic" }
+          },
+          {
+            "icon": null,
+            "name": { "key": "SplitPaneVerticalName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "vertical" }
+          },
+          {
+            "icon": null,
+            "name": { "key": "SplitPaneHorizontalName" },
+            "command": { "action": "splitPane", "profile": "${profile.name}", "split": "horizontal" }
+          }
+        ]
+      }
+    ]
+  }
+]
+```
+
+A complete diagram of what the default Command Palette will look like given the
+default keybindings and these changes is given in [**Appendix
+2**](#appendix-2-complete-default-command-palette).
+
+## Concerns
+
+**TODO: FOR DISCUSSION**: "New tab with index {index}". How does this play with
+the new tab dropdown customizations in [#5888]? In recent iterations of that
+spec, we changed the meaning of `{ "action": "newTab", "index": 1 }` to mean
+"open the first entry in the new tab menu". If that's a profile, then we'll open
+a new tab with it. If it's an action, we'll perform that action. If it's a
+nested menu, then we'll open the menu to that entry.
+
+Additionally, how exactly does that play with something like `{ "action":
+"newTab", "index": 1, "commandline": "wsl.exe" }`? This is really a discussion
+for that spec, but is an issue highlighted by this spec. If the first entry is
+anything other than a `profile`, then the `commandline` parameter doesn't really
+mean anything anymore. I'm tempted to revert this particular portion of the new
+tab menu customization spec over this.
+
+We could instead add an `index` to `openNewTabDropdown`, and have that string
+instead be "Open new tab dropdown, index:1". That would help disambiguate the
+two.

```

Comment:
I think this may be the best thing to do.

We'll still have to deal with the "newTab" actions we have that iterate on each index though :/

Reply:
I'm really _really_ in favor of this option, because it's way clearer IMO.

Reply:
Yeah, I agree. `newTab` is profile-list-relative.